### PR TITLE
NIFI-4377: Added a fieldName() function to RecordPath and addressed a…

### PR DESCRIPTION
**Pull request by markap14**
…n issue that caused //* to not work

Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [ ] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by pvillard31** [comment]
reviewing

File: nifi-commons/nifi-record-path/src/main/java/org/apache/nifi/record/path/paths/WildcardDescendantPath.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    private List<FieldValue> findDescendants(final FieldValue fieldValue) {
+        if (fieldValue == null || fieldValue.getValue() == null) {
+            return Collections.emptyList();
+        }
+        if (!Filters.isRecord(fieldValue)) {
+            return Collections.emptyList();
+        }
+
+        final Record record = (Record) fieldValue.getValue();
```
[the remark this answers is not in the document]
Comment (markap14) [reply nifi-commons/nifi-record-path/src/main/java/org/apache/nifi/record/path/paths/WildcardDescendantPath.java]: Actually I think this is addressed already online 47:
```
if (fieldValue == null || fieldValue.getValue() == null) {
    return Collections.emptyList();
}
```

**Comment by markap14** [comment]
@pvillard31 I pushed a new commit to update the Record Path Guide. I also added to the unit tests for the fieldName() function

File: nifi-commons/nifi-record-path/src/main/java/org/apache/nifi/record/path/paths/WildcardDescendantPath.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (pvillard31) [reply nifi-commons/nifi-record-path/src/main/java/org/apache/nifi/record/path/paths/WildcardDescendantPath.java]: Right!

**Pull request closed without merging**